scoutfs-utils changes for recent alignment work

agrover opened the pull request:
The padding stuff and then also some compiler warnings using gcc 10.

Commit:
scoutfs-utils: avoid redeclarations of __[be,le][16,32,64] in sparse.h

dev.c includes linux/fs.h which includes linux/types.h, which defines
these types, __be16 etc. These are also defined in sparse.h, but I don't
think these are needed.

Definitions in linux/types.h includes stuff to set attr(bitwise) if
__CHECKER__ is defined, so we can remove __sp_biwise.

Signed-off-by: Andy Grover <agrover@versity.com>

Commit:
scoutfs-utils: specify types for some long constants

This avoids warnings on Centos 7 from gcc 4.8.5.

zabbo committed:
scoutfs-utils: remove btree item owner

We no longer have an owner offset trailing btree item values.

Signed-off-by: Zach Brown <zab@versity.com>

zabbo reviewed (approved):
This all seems great, thanks!  We'll merge this once the -kmod side is ready.

Commit:
scoutfs-utils: Sync latest headers with kernel code

__packed no longer used.

Signed-off-by: Andy Grover <agrover@versity.com>

Commit:
scoutfs-utils: Enable -Wpadded

The compiler will complain if it sees any padding.

Fix a spot in print.c for this.

Signed-off-by: Andy Grover <agrover@versity.com>
